### ISSUE TITLE
Stop ignoring CFLAGS request

### ISSUE THREAD

**Issue by jengelh** [openucx/ucx#1547]
```
export CFLAGS=-Wno-error
export CXXFLAGS=-Wno-error
./configure CFLAGS=-Wno-error CXXFLAGS=-Wno-error
[   21s] + make -j8 V=1
[   21s] make  all-recursive
[   21s] make[1]: Entering directory '/home/abuild/rpmbuild/BUILD/ucx-0~git2976'
[   21s] Making all in src/ucm
[   21s] make[2]: Entering directory '/home/abuild/rpmbuild/BUILD/ucx-0~git2976/src/ucm'
[   21s] /bin/sh ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../..  -I/home/a
build/rpmbuild/BUILD/ucx-0~git2976/src -I/home/abuild/rpmbuild/BUILD/ucx-0~git2976/src -DUCM_MALLOC_
PREFIX=ucm_dl  -fno-strict-aliasing -DUSE_LOCKS=1 -DMALLINFO_FIELD_TYPE=int -DCPU_FLAGS="|avx" -Wno-
deprecated-declarations -O3 -g -Wall -Werror  -mavx -c -o event/libucm_la-event.lo `test -f 'event/e
vent.c' || echo './'`event/event.c
```
You *must not* override CFLAGS within configure.ac, or any files included from there. Or if you do, it must be restored at the end of it.

**Comment by shamisp** [comment]
👍 @yosefe I noticed the same. This is very wrong ...

**Comment by MattBBaker** [comment]
I actually looked into this and found the problem. Posted a solution in #1554

**Comment by shamisp** [comment]
@MattBBaker Awesome. please open PR for master as well (in addition to v1.2)

**Comment by MattBBaker** [comment]
@shamisp I will, just wanted to make sure nothing unexpectedly exploded before opening 2 PRs.

**Comment by shamisp** [comment]
Ok

**Comment by shamisp** [comment]
This one was officially resolved by #1555 and #1554
